src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    private final AbstractFigure[] figures = new AbstractFigure[]{
```

Comment:
This defeats the purpose of the class: this array is filled only once. This is not a truly random generator/supplier class. In your previous implementation, a new figure was generated each time the method was triggered, now you are randomizing a type of figure, but not the Figure itself, in other words, if you get a RightTriangle, it will always be the same. Recover your previous version, and move all private members after public.